### PR TITLE
Refreshed Bengali Translations and Added New Translations to Bengali

### PR DESCRIPTION
**Pull request by abhik-99**
This PR aims to do the following:
1. Refresh the Bengali Translations
2. Add new translations in line with #653 to complete up to 10%.
3. Corrections to existing translations.

Edit: Recent commits bump up the % completed to 12.

### CODE REVIEW

**Comment by mgeisler** [comment]
Hi @abhik-99, thanks a lot for this!

I took a quick look at the PR using

```
gh pr diff 1876 | bat -l patch
```

and I see that all the paths separators have changed from `/` to `\`:


```patch
-#: src/SUMMARY.md src/index.md
+#: src\SUMMARY.md src\index.md
 msgid "Welcome to Comprehensive Rust 🦀"
 msgstr "Comprehensive Rust 🦀 এ আপনাকে স্বাগতম"
 
-#: src/SUMMARY.md src/running-the-course.md
+#: src\SUMMARY.md src\running-the-course.md
 msgid "Running the Course"
 msgstr "কোর্স চালানো"
 
-#: src/SUMMARY.md src/running-the-course/course-structure.md
+#: src\SUMMARY.md src\running-the-course/course-structure.md
 msgid "Course Structure"
 msgstr "কোর্স এর গঠন"
```

It's the first time that has happened — and I think we have other translators who use Windows... perhaps there is a setting for this in Poedit?

In any case, I hope you can do a big search-and-replace and turn them back into normal slashes. The diff will still be huge, but this will at least remove some unnecessary changes.

**Comment by mgeisler** [comment]
Also, you should avoid mixing refreshing with actual translation work (as mentioned [briefly in the guide](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md#refreshing-an-existing-translation)).

I hope the reviewers can skim through the new translations and then we merge this PR — the next translations will be much smaller since there will be no refresh step.

**Comment by abhik-99** [comment]
> Hi @abhik-99, thanks a lot for this!
> 
> I took a quick look at the PR using
> 
> ```
> gh pr diff 1876 | bat -l patch
> ```
> 
> and I see that all the paths separators have changed from `/` to `\`:
> 
> ```diff
> -#: src/SUMMARY.md src/index.md
> +#: src\SUMMARY.md src\index.md
>  msgid "Welcome to Comprehensive Rust 🦀"
>  msgstr "Comprehensive Rust 🦀 এ আপনাকে স্বাগতম"
>  
> -#: src/SUMMARY.md src/running-the-course.md
> +#: src\SUMMARY.md src\running-the-course.md
>  msgid "Running the Course"
>  msgstr "কোর্স চালানো"
>  
> -#: src/SUMMARY.md src/running-the-course/course-structure.md
> +#: src\SUMMARY.md src\running-the-course/course-structure.md
>  msgid "Course Structure"
>  msgstr "কোর্স এর গঠন"
> ```
> 
> It's the first time that has happened — and I think we have other translators who use Windows... perhaps there is a setting for this in Poedit?
> 
> In any case, I hope you can do a big search-and-replace and turn them back into normal slashes. The diff will still be huge, but this will at least remove some unnecessary changes.

Yes, I use Windows. Will search for this separator setting. Meanwhile, I did a find + replace of the paths. Hope the test succeeds. Thanks for the heads up!

**Comment by abhik-99** [comment]
Also, it seems that running `dprint fmt po/bn.po` in my system might have changed the formatting for some reason even though I haven't configured anything custom in dprint.

File: po/bn.po [hunk 1]
```diff
@@ -56,38 +56,37 @@ msgstr "প্রথম দিনঃ সকাল"
...
+msgstr "হ্যালো ওয়ার্ল্ড"
 
 #: src/SUMMARY.md src/hello-world/what-is-rust.md
 msgid "What is Rust?"
 msgstr "Rust কি?"
 
 #: src/SUMMARY.md src/hello-world/benefits.md
 msgid "Benefits of Rust"
-msgstr ""
+msgstr "Rust এর সুবিধা"
```
Comment (mirwasi) [inline po/bn.po:69]:
```suggestion
msgstr "রাস্ট এর সুবিধা"
```
Let's not mix Bengali and English terms.

File: po/bn.po [hunk 2]
```diff
@@ -56,38 +56,37 @@ msgstr "প্রথম দিনঃ সকাল"
...
 
 #: src/SUMMARY.md src/types-and-values/values.md
 msgid "Values"
-msgstr ""
+msgstr "মান"
 
 #: src/SUMMARY.md src/types-and-values/arithmetic.md
 msgid "Arithmetic"
-msgstr ""
+msgstr "অঙ্কশাস্ত্র"
```
Comment (mirwasi) [inline po/bn.po:89]: গণনা or হিসাব seems more appropriate.

File: po/bn.po [hunk 3]
```diff
@@ -111,42 +110,56 @@ msgstr ""
...
-msgid "Conditionals"
-msgstr ""
+#: src/SUMMARY.md
+msgid "`if` Expressions"
+msgstr "if অভিব্যক্তি"
 
 #: src/SUMMARY.md src/control-flow-basics/loops.md
 msgid "Loops"
-msgstr ""
+msgstr "লুপ্স (বা ফাঁস)"
```
Comment (mirwasi) [inline po/bn.po:126]:
```suggestion
msgstr "লুপ্স"
```

 ফাঁস doesn't sound appropriate.

File: po/bn.po [hunk 4]
```diff
@@ -111,42 +110,56 @@ msgstr ""
...
+msgstr "`loop`"
 
 #: src/SUMMARY.md src/control-flow-basics/break-continue.md
 msgid "`break` and `continue`"
-msgstr ""
+msgstr "`break` এবং  `continue`"
+
+#: src/SUMMARY.md src/control-flow-basics/break-continue/labels.md
+msgid "Labels"
+msgstr "আখ্যা"
```
Comment (mirwasi) [inline po/bn.po:142]:
```suggestion
msgstr "লেবেল"
```
Let's use 'label' here.

**Review by mirwasi — APPROVED** [review]
Thanks for your contribution,  @abhik-99!

I made some comments, however I won't stop this from merging. Once we have enough translations, we can focus on the readability.

File: po/bn.po [hunk 5]
```diff
@@ -213,165 +216,169 @@ msgid "Enums"
...
-
-#: src/SUMMARY.md src/methods-and-traits/trait-objects.md
-msgid "Trait Objects"
-msgstr "Trait Objects"
+msgstr "আহরণ করা"
 
 #: src/SUMMARY.md src/methods-and-traits/exercise.md
 msgid "Exercise: Generic Logger"
-msgstr ""
+msgstr "অনুশীলনী: জাতীয় তথ্য নির্মাতা"
```
[the remark this answers is not in the document]
Comment (abhik-99) [reply po/bn.po:280]: No worries, will add this modification manually along with the others.

File: po/bn.po [hunk 6]
```diff
@@ -155,53 +168,43 @@ msgstr "প্রথম দিনঃ বিকাল"
...
-#, fuzzy
 msgid "Shared References"
-msgstr "রেফারেন্স"
+msgstr "ভাগ করা উল্লেখ"
 
 #: src/SUMMARY.md src/references/exclusive.md
-#, fuzzy
 msgid "Exclusive References"
-msgstr "ড্যাঙ্গলিং রেফারেন্স"
+msgstr "একচেটিয়া উল্লেখ"
```
[the remark this answers is not in the document]
Comment (abhik-99) [reply po/bn.po:195]: Sure, will do a find and replace for it all.

File: po/bn.po (hunk 2 above)
Comment (abhik-99) [reply po/bn.po:89]: @mirwasi how about গণিত ?

**Comment by mgeisler** [comment]
> Yes, I use Windows. Will search for this separator setting. Meanwhile, I did a find + replace of the paths. Hope the test succeeds. Thanks for the heads up!

Thanks, the paths look good now.

Perhaps nobody have tried running `dprint fmt` on this yet on Windows... I'm afraid I don't know. If this becomes a problem again, then please open a separate issue so we can investigate it more.

**Comment by mgeisler** [comment]
> I made some comments, however I won't stop this from merging. Once we have enough translations, we can focus on the readability.

I think that's a great approach! This PR should really just have been a clean refresh, and then changes can be made in smaller more focused PRs afterwards. So I'm happy to get this merged now and then you can improve the language in future iterations.

File: po/bn.po (hunk 1 above)
Comment (mgeisler) [reply po/bn.po:69]: > I saw the Japanese translation had kept "Rust" untranslated as a word which is why I was confused.

It's a really good question what we should do with names like "Rust" in languages that don't use Latin letters. I don't know myself what the custom is, so we've simply let it be up to the taste of the individual translators.

I would be interested in having a discussion about this, so feel free to raise the question in https://github.com/google/comprehensive-rust/discussions (if you want).

**Comment by mgeisler** [comment]
> I made some comments, however I won't stop this from merging.

@mirwasi, I'll let you click the "Squash and merge" button when you feel this is ready — tests pass now so it's ready to go in.

File: po/bn.po (hunk 1 above)
Comment (abhik-99) [reply po/bn.po:69]: I have started this discussion [here](https://github.com/google/comprehensive-rust/discussions/1888). Meanwhile, I am taking @mirwasi 's advice and translating it to Bengali with the same pronounciation. Would change it depending upon the consensus of the discussion.

**Pull request merged**